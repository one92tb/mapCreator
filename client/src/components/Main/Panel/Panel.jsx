import React, { Component } from "react";
import MarkerList from "./MarkerList/MarkerList";
import { isPanelSelect } from "../../../actions/isPanelSelect";
import { connect } from "react-redux";
import PropTypes from "prop-types";
import {
  Wrapper,
  FilterLink,
  SelectLink,
  Card,
  CardHeader,
  CardBody,
  NavItem,
  Nav
} from "./style";

class Panel extends Component {
  constructor(props) {
    super(props);
    this.state = {
      isSelected: true
    };
  }

  componentDidUpdate(prevProps) {
    const { location } = this.props;
    if (
      location.pathname !== prevProps.location.pathname &&
      location.pathname === "/createMarker"
    ) {
      this.setState({
        isSelected: true
      });
    }
  }

  switch = bool => {
    const { isSelected } = this.props;
    this.setState({
      isSelected: bool
    });

    isPanelSelect(isSelected);
  };

  render() {
    const { isSelected } = this.state;
    return (
      <Wrapper>
        <Card>
          <CardHeader>
            <Nav>
              <NavItem>
                <SelectLink
                  exact={true}
                  isSelected={isSelected}
                  onClick={() => this.switch(true)}
                >
                  Select marker
                </SelectLink>
              </NavItem>
              /
              <NavItem>
                <FilterLink
                  isSelected={isSelected}
                  location={this.props.location.pathname}
                  onClick={() => this.switch(false)}
                >
                  Filter marker
                </FilterLink>
              </NavItem>
            </Nav>
          </CardHeader>
          <CardBody className="scroll">
            <MarkerList isNavSelect={isSelected} />
          </CardBody>
        </Card>
      </Wrapper>
    );
  }
}

const mapDispatchToProps = {
  isPanelSelect
};

export default connect(
  null,
  mapDispatchToProps
)(Panel);

Panel.propTypes = {
  isPanelSelect: PropTypes.func.isRequired
};
