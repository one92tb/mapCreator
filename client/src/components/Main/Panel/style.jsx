import { css } from "styled-components";
import styled from "styled-components";

const Wrapper = styled.div`
  background:#f2f2f2
  padding: 40px 10px 40px 20px;
  height: 100%;
`;

const NavLink = css`
  margin: 0 10px;
  cursor: pointer;
`;

//  pointer-events: none;
const FilterLink = styled.a`
  ${NavLink};

  ${({ isSelected }) =>
    !isSelected &&
    `
    font-weight: bold;
  `};

  ${({ location }) =>
    location === "/createMarker" &&
    `
    pointer-events: none;
    color: #808080!important;
  `};
`;

const SelectLink = styled.a`
  ${NavLink};

  ${({ isSelected }) =>
    isSelected &&
    `
    font-weight: bold
  `};
`;

const Card = styled.div`
  height: 100%;
  border: 1px solid #00b8e6;
  width: 100%;
  position: relative;
  display: flex;
  min-width: 0;
  word-wrap: break-word;
  flex-direction: column;
  border-radius: 3px;
  overflow: hidden;
`;

const CardHeader = styled.div`
  border-bottom: 1px solid #00b8e6;
  padding: 0.75rem 1.25rem;
  background: #4ddbff;
  display: block;
`;

const Nav = styled.ul`
  display: flex;
  flex-wrap: wrap;
  list-style: none;
  padding-left: 0;
  margin-bottom: 0;
  justify-content: center;
`;
const NavItem = styled.li``;

const CardBody = styled.div`
  padding: 1.25rem;
  height: calc(100% - 48px);
  overflow-y: auto;
`;

export {
  Wrapper,
  FilterLink,
  SelectLink,
  Card,
  CardHeader,
  CardBody,
  NavItem,
  Nav
};
