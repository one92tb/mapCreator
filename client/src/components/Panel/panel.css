.panelCard{
  height: 100%;
}
.scroll {
    min-height: 100px;
    overflow-y: auto;
}
