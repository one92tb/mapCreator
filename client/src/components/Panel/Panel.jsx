import React, { Component } from "react";
import { Card, CardHeader, CardBody, Nav, NavItem, NavLink } from "reactstrap";
import "./panel.css";
import MarkerList from "./MarkerList/MarkerList";
import MarkerFilter from "./MarkerFilter/MarkerFilter";
import { Link, Route, Switch } from "react-router-dom";

const routes = [
  {
    path: "/",
    exact: true,
    sidebar: () => <MarkerList />
  },
  {
    path: "/createMarker",
    sidebar: () => <MarkerList />
  },
  {
    path: "/selectMarker",
    sidebar: () => <MarkerList />
  },
  {
    path: "/filterMarker",
    sidebar: () => <MarkerFilter />
  }
];

class Panel extends Component {
  render() {
    return (
      <Card className="panelCard">
        <CardHeader>
          <Nav>
            <NavItem>
              <NavLink tag={Link} to="/selectMarker">
                Select Marker
              </NavLink>
            </NavItem>
            <NavItem>
              <NavLink tag={Link} to="/filterMarker">
                Fileter Marker
              </NavLink>
            </NavItem>
          </Nav>
        </CardHeader>
        <CardBody className="scroll">
          <Switch>
            {routes.map((route, id) => (
              <Route
                key={id}
                path={route.path}
                exact={route.exact}
                component={route.sidebar}
              />
            ))}
          </Switch>
        </CardBody>
      </Card>
    );
  }
}

export default Panel;
