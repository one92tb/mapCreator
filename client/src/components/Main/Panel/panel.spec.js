import React from "react";
import { shallow, mount } from "enzyme";
import toJson from "enzyme-to-json";
import configureStore from "redux-mock-store";
import { Panel } from "./Panel";
import history from "../../../history";

describe("panel component", () => {
  let wrapper;

  it("passess all props to panel component", () => {
    const isPanelSelect = jest.fn();
    const history = {
      location: {
        pathname: "/",
        search: "",
        hash: "",
        state: undefined,
        key: "pncigm"
      }
    };

    wrapper = shallow(<Panel isPanelSelect={isPanelSelect} {...history} />);
    expect(wrapper).toMatchSnapshot();
  });

  it("component did update - change location", () => {
    const isPanelSelect = jest.fn();
    const history = {
      location: {
        pathname: "/",
        search: "",
        hash: "",
        state: undefined,
        key: "pncigm"
      }
    };

    wrapper = shallow(<Panel isPanelSelect={isPanelSelect} {...history} />);

    wrapper.setState({
      isSelected: false
    });
    wrapper.setProps({
      location: {
        pathname: "/createMarker",
        search: "",
        hash: "",
        state: undefined,
        key: "pncigm"
      }
    });

    expect(wrapper.state().isSelected).toBe(true);
  });

  it("handleCheckBox method - change checked in state", () => {
    const isPanelSelect = jest.fn();
    const history = {
      location: {
        pathname: "/",
        search: "",
        hash: "",
        state: undefined,
        key: "pncigm"
      }
    };

    wrapper = shallow(<Panel isPanelSelect={isPanelSelect} {...history} />);

    const input = wrapper.find("input");

    input.simulate("change", {
      target: {
        checked: true
      }
    });

    expect(wrapper.state().checked).toBe(true);

    input.simulate("change", {
      target: {
        checked: false
      }
    });

    expect(wrapper.state().checked).toBe(false);
  });

  it("switch method - change select marker to filter marker and vice versa", () => {
    const isPanelSelect = jest.fn();
    const history = {
      location: {
        pathname: "/",
        search: "",
        hash: "",
        state: undefined,
        key: "pncigm"
      }
    };

    wrapper = shallow(<Panel isPanelSelect={isPanelSelect} {...history} />);

    const selectLink = wrapper.find("a").first();
    const filterLink = wrapper.find("a").at(1);

    selectLink.simulate("click", true);
    expect(wrapper.state().isSelected).toBe(true);
    filterLink.simulate("click", false);
    expect(wrapper.state().isSelected).toBe(false);
    expect(isPanelSelect).toHaveBeenCalledTimes(2);
  });
});
