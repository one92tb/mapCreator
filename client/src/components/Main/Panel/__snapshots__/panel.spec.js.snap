// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`panel component passess all props to panel component 1`] = `
<div
  currentLocation={
    Object {
      "hash": "",
      "key": "pncigm",
      "pathname": "/",
      "search": "",
      "state": undefined,
    }
  }
>
  <label
    currentLocation={
      Object {
        "hash": "",
        "key": "pncigm",
        "pathname": "/",
        "search": "",
        "state": undefined,
      }
    }
    htmlFor="panel"
  >
    <img
      alt="drawMarker"
      height={30}
      src="drawMarker.png"
      width={30}
    />
  </label>
  <input
    id="panel"
    onChange={[Function]}
    type="checkbox"
  />
  <div
    currentLocation={
      Object {
        "hash": "",
        "key": "pncigm",
        "pathname": "/",
        "search": "",
        "state": undefined,
      }
    }
    isChecked={false}
  >
    <div>
      <ul>
        <li>
          <a
            exact={true}
            isSelected={true}
            onClick={[Function]}
          >
            Select marker
          </a>
        </li>
        /
        <li>
          <a
            isSelected={true}
            location="/"
            onClick={[Function]}
          >
            Filter marker
          </a>
        </li>
      </ul>
    </div>
    <div
      className="scroll"
    >
      <Connect(MarkerList)
        isNavSelect={true}
      />
    </div>
  </div>
</div>
`;
